src/middleware/arns.ts

```diff
@@ -123,7 +123,21 @@ export const createArnsMiddleware = ({
         name: arnsSubdomain,
       });
     end();
-    if (resolvedId === undefined) {
+    if (resolvedId !== undefined) {
+      // Populate the ArNS name headers if resolution was successful
+      res.header(headerNames.arnsName, arnsSubdomain);
+      if (arnsSubdomain) {
+        const parts = arnsSubdomain.split('_');
+        const basename = parts.pop(); // last part is basename
+        const undername = parts.join('_'); // everything else is undername
+        if (basename !== undefined && basename !== '') {
+          res.header(headerNames.arnsBasename, basename);
+        }
+        if (undername !== undefined && undername !== '') {
+          res.header(headerNames.arnsUndername, undername);
+        }
+      }
```

Comment:
nit: the resolver could return these values vs. computing them again here

Reply:
Hmm... true. I'll check on patching this up.